src/python/WMCore/Services/LogDB/LogDB.py

```diff
@@ -126,5 +127,55 @@ def cleanup(self, thr, backend='local'):
         try:
             self.backend.cleanup(thr)
         except Exception as exc:
-            self.logger.error('LogDBBackend cleanup API failed, backend=%s, error=%s' \
-                    % (backend, str(exc)))
+            self.logger.error('LogDBBackend cleanup API failed, backend=%s, error=%s', backend, str(exc))
+    
+    def heartbeat_report(self):
+        report = defaultdict(dict)
+        if self.user_pat.match(self.identifier):
+            self.logger.error("User %s: doesn't allow this function", self.identifier) 
+            return report
+        
+        for row in self.backend.get(self.default_user, None, agent=True).get('rows', []):
+            identifier = row['doc']['identifier']
+            # wmstats thread can run in multiple boxed. 
+            if self.identifier == identifier or identifier.startswith(self.identifier):
+                #this will handle wmstats DataCacheUpdate thread with multiple machine
+                postfix = identifier.replace(self.identifier, "")
```

Comment:
This is the special case I had to handle. In wmstats, DataCacheUpdate thread not just run in one threads but all the threads reqmon was deployed. It might be just one reqmon instance now but we can't assume that future either.
In that case I added the postfix with host name to distinguish different thread in different box. (I still need to improve the detection if one thread is not running in one machine though - it won't detect if it never started.)